components/tag-list/tag-list-item.js

```diff
@@ -9,12 +10,46 @@ class TagListItem extends TagListItemMixin(LitElement) {
 			 * REQUIRED: Text to display
 			 * @type {string}
 			 */
-			text: { type: String }
+			text: { type: String },
+			/**
+			 * Optional: Text to display in tooltip.
+			 * Tooltip will also include text property value if truncated.
+			 * @type {string}
+			 */
+			description: { type: String },
+			_isTruncated: { attribute: false }
```

Comment:
The Lit 2 way to do this is with state -- all the benefits of properties, but they're not settable by consumers.

```suggestion
			_isTruncated: { state: true }
```